setup-dev.sh

```
@@ -17,6 +18,125 @@ print_error() {
+    print_warning "mediasoup worker directory not found. Skipping compilation."
+    return 0
+  fi
+  
+  # Check if worker binary already exists
+  if [ -f "$worker_dir/out/Release/mediasoup-worker" ] || [ -f "$worker_dir/out/Release/mediasoup-worker.exe" ]; then
+    print_success "mediasoup worker binary already exists"
+    return 0
+  fi
+  
+  print_step "Compiling mediasoup worker..."
+  cd "$worker_dir"
```

Shellcheck (0.10.0)

[warning] Use 'cd ... || exit' or 'cd ... || return' in case cd fails.

(SC2164)